packages/extension-chakra-store-locator/src/components/with-optional-commerce-sdk-react-provider.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import {getAppOrigin} from '@salesforce/pwa-kit-react-sdk/utils/url'
+import {CommerceApiProvider, useCommerceApi} from '@salesforce/commerce-sdk-react'
+import {UserConfig} from '../types/config'
+
+/**
+ * Checks if the CommerceApiProvider is already installed in the component tree.
+ * @returns boolean, true if the CommerceApiProvider is installed, false otherwise.
+ */
+const useHasCommerceApiProvider = () => {
+    let hasProvider = false
+
+    try {
+        const api = useCommerceApi()
+
+        // the api object is an object with a bunch of api clients like ShopperProduct, ShopperOrder, etc.
+        // if the object is empty, then the CommerceApiProvider is not installed
+        if (Object.keys(api).length > 0) {
+            hasProvider = true
+        }
+    } catch (_) {
+        hasProvider = false
+    }
+
+    return hasProvider
+} 
+
+type WithOptionalCommerceSdkReactProvider = React.ComponentPropsWithoutRef<any>
+
+/**
+ * Higher-order component that conditionally installs the CommerceApiProvider if the config is provided.
+ *
+ * @param WrappedComponent - The component to be optionally wrapped with CommerceApiProvider.
+ * @param config - The configuration object for the CommerceApiProvider.
+ * @returns A component that wraps the given component with CommerceApiProvider if it is not already present in the component tree.
+ */
+export const withOptionalCommerceSdkReactProvider = <P extends object>(
+    WrappedComponent: React.ComponentType<P>,
+    config: UserConfig
+) => {
+    const HOC: React.FC<P> = (props: WithOptionalCommerceSdkReactProvider) => {
+        if (useHasCommerceApiProvider()) {
+            return <WrappedComponent {...(props as P)} />
+        }
+        if (!config.commerceApi || !config.commerceApi?.parameters) {
+            return <WrappedComponent {...(props as P)} />
+        }
```

Comment:
Should we be we be warning or erroring out don't have a configuration supplied instead of not using the provider. I think thing would end up erroring out and that error wouldn't be as close to the source as this warning/error.

Reply:
added a `logger.error`